Top100/top100songs.py

```diff
@@ -0,0 +1,257 @@
+"""This code finds the top 100 most popular songs by parsing and without using OOP
+the HOT 100 index by Billboard (https://www.billboard.com/charts/hot-100/)
+plots the 100 most used words in all these songs lyrics
+and shows the top 10 artists that use the most words"""
+
+import string
+import matplotlib.pyplot as plt
+import requests
+
+from bs4 import BeautifulSoup as Soup
+from typing import Union
+
+
+def get_content_from_url(url: str) -> Soup:
+    """ This function requests the content of a given HTML url,
+     parses it and returns a list of all relevant content under the text_specifier.
+    :param url: str
+    :return: list
+    """
+    # Connect to url and check response
+    req = requests.get(url)
+    if req.status_code != 200:
+        print(url)
+        raise ConnectionError
+
+    # Get content from artist page
+    return Soup(req.text, 'html.parser')
+
+
+def find_song_title_link(song_title: str, songs: list) -> str:
+    """This function iterates through HTML syntax from https://www.lyrics.com/ website
+    parsed by the get_content_from_url function
+    and find's the relative link to the song_title lyrics.
+    :param song_title: str
+    :param songs: list of HTML syntax content
+    :return: str, relative path to url
+    """
+    for song in songs:
+        find_song_title = song.a.text.lower()
+        if '(' in find_song_title:
+            find_song_title = find_song_title.split(' (')[0]
+        if song_title.lower() == find_song_title:
+            return song.find('a').attrs['href']
+    return ''
+
+
+def get_full_url_to_lyrics(song_title: str, artist: str) -> str:
```

Comment:
The funcion is way to lomg amd complex. Refator it.

Reply:
did my best, added few functions to shorten it.